app/models/piece.rb

```diff
@@ -158,6 +139,18 @@ def is_white?(piece)
  		piece.color == "white" ? true : false
  	end	
 
+ 	def puts_in_check?(x,y)
+		if color == "white"
+			opposite_color = "black"
+		elsif color == "black"
+			opposite_color = "white"
+		end
+				
+		opponents = game.pieces.where(color: opposite_color)
+		opponents.each do |piece|
+			return false unless piece.valid_move?(x,y) 
```

Comment:
@zoheb391 There's no break statement, but I think this code would function the same way. It's also a little shorter.